spec/models/authenticator_spec.rb

```diff
@@ -25,15 +25,15 @@
     end
 
     it 'Updates the user groups when no matching groups' do
-      expect(authenticator).to receive(:find_external_identity)
+      allow(authenticator).to receive(:find_external_identity).with(any_args)
```

Comment:
The `allows` seem to weaken the spec to me.

I also think rubocop will complain about alignment of this. I think you need this:
```ruby
      it 'Updates the user groups when no matching groups' do
        allow(authenticator)
          .to receive(:find_external_identity)
          .with(any_args)
          .and_return([{:username => user.userid, :fullname => user.name, :domain => "example.com"}, []])
  
        authenticator.authorize(task.id, user.userid)
        expect(user.reload.miq_groups).to be_empty
      end
  
      it 'Updates the user groups' do
        allow(authenticator)
          .to receive(:find_external_identity)
          .with(any_args)
          .and_return([{:username => user.userid, :fullname => user.name, :domain => "example.com"}, groups.collect(&:name)])
  
        authenticator.authorize(task.id, user.userid)
        expect(user.reload.miq_groups).to match_array(groups)
      end
```

Reply:
The question is whether or not that's an actual expectation. It sure doesn't look like it to me, but just in case I'm wrong, I've reverted that change.

Reply:
Regarding rubocop, I ran `rubocop -a` and applied the changes. There are currently no rubocop violations now.